### ISSUE TITLE
[🐛 Bug]: Unable to run tests with BiDi and Saucelabs

### ISSUE THREAD

**Issue by thamp59** [webdriverio/webdriverio#13839]
### Have you read the Contributing Guidelines on issues?

- [X] I have read the [Contributing Guidelines on issues](https://github.com/webdriverio/webdriverio/blob/main/CONTRIBUTING.md#reporting-new-issues).

### WebdriverIO Version

9.2.4

### Node.js Version

20.18.0

### Mode

WDIO Testrunner

### Which capabilities are you using?

```typescript
{
  browserName: 'chrome',
  // webSocketUrl: true // This also fails on late versions of v8
  'sauce:options': {
    tags: ['Chrome-Windows-10']
  }
}
```

### What happened?

When trying to connect with sauce labs using WebDriver BiDi, I run into a connection error.

[0-0] 2024-10-30T18:41:49.188Z INFO webdriver: Register BiDi handler for session with id 57e760490965499a9f78f5e13764234a
[0-0] 2024-10-30T18:41:49.188Z INFO webdriver: Connect to webSocketUrl ws://127.0.0.1:21365/session/d362cc3d627a3738764efa39d873185a
[0-0] node:events:497
[0-0]       throw er; // Unhandled 'error' event
[0-0]       ^
[0-0]
[0-0] Error: connect ECONNREFUSED 127.0.0.1:21365
[0-0]     at TCPConnectWrap.afterConnect [as oncomplete] (node:net:1607:16)
[0-0] Emitted 'error' event on WebSocket instance at:
[0-0]     at emitErrorAndClose (C:\dev\wdio-iframe-elements-thamp\node_modules\ws\lib\websocket.js:1041:13)
[0-0]     at ClientRequest.<anonymous> (C:\dev\wdio-iframe-elements-thamp\node_modules\ws\lib\websocket.js:881:5)
[0-0]     at ClientRequest.emit (node:events:519:28)
[0-0]     at ClientRequest.emit (node:domain:488:12)
[0-0]     at emitErrorEvent (node:_http_client:101:11)
[0-0]     at Socket.socketErrorListener (node:_http_client:504:5)
[0-0]     at Socket.emit (node:events:519:28)
[0-0]     at Socket.emit (node:domain:488:12)
[0-0]     at emitErrorNT (node:internal/streams/destroy:169:8)
[0-0]     at emitErrorCloseNT (node:internal/streams/destroy:128:3) {
[0-0]   errno: -4078,
[0-0]   code: 'ECONNREFUSED',
[0-0]   syscall: 'connect',
[0-0]   address: '127.0.0.1',
[0-0]   port: 21365
[0-0] }
[0-0]
[0-0] Node.js v20.18.0

### What is your expected behavior?

Tests should run on saucelabs without issue.

### How to reproduce the bug.

https://github.com/thamp59/wdio-iframe-elements/pull/2

### Relevant log output

```typescript
$ npm run wdio

> wdio-iframe-elements@1.0.0 wdio
> wdio run wdio.conf.js


Execution of 1 workers started at 2024-10-30T18:41:34.679Z

2024-10-30T18:41:34.754Z INFO @wdio/cli:launcher: Run onPrepare hook
2024-10-30T18:41:34.756Z INFO @wdio/cli:launcher: Run onWorkerStart hook
2024-10-30T18:41:34.757Z INFO @wdio/local-runner: Start worker 0-0 with arg: run wdio.conf.js
[0-0] 2024-10-30T18:41:35.408Z INFO @wdio/local-runner: Run worker command: run
[0-0] RUNNING in chrome - file:///C:/dev/wdio-iframe-elements-thamp/smallTest.spec.js
[0-0] 2024-10-30T18:41:35.564Z INFO webdriver: Initiate new session using the WebDriver protocol
[0-0] 2024-10-30T18:41:35.564Z INFO @wdio/utils: Connecting to existing driver at https://ondemand.us-west-1.saucelabs.com:443/wd/hub
[0-0] 2024-10-30T18:41:35.571Z INFO webdriver: [POST] https://ondemand.us-west-1.saucelabs.com/wd/hub/session
[0-0] 2024-10-30T18:41:35.571Z INFO webdriver: DATA {
[0-0]   capabilities: {
[0-0]     alwaysMatch: {
[0-0]       browserName: 'chrome',
[0-0]       'sauce:options': [Object],
[0-0]       webSocketUrl: true,
[0-0]       unhandledPromptBehavior: 'ignore'
[0-0]     },
[0-0]     firstMatch: [ {} ]
[0-0]   }
[0-0] }
[0-0] 2024-10-30T18:41:49.188Z INFO webdriver: Register BiDi handler for session with id 57e760490965499a9f78f5e13764234a
[0-0] 2024-10-30T18:41:49.188Z INFO webdriver: Connect to webSocketUrl ws://127.0.0.1:21365/session/d362cc3d627a3738764efa39d873185a
[0-0] node:events:497
[0-0]       throw er; // Unhandled 'error' event
[0-0]       ^
[0-0]
[0-0] Error: connect ECONNREFUSED 127.0.0.1:21365
[0-0]     at TCPConnectWrap.afterConnect [as oncomplete] (node:net:1607:16)
[0-0] Emitted 'error' event on WebSocket instance at:
[0-0]     at emitErrorAndClose (C:\dev\wdio-iframe-elements-thamp\node_modules\ws\lib\websocket.js:1041:13)
[0-0]     at ClientRequest.<anonymous> (C:\dev\wdio-iframe-elements-thamp\node_modules\ws\lib\websocket.js:881:5)
[0-0]     at ClientRequest.emit (node:events:519:28)
[0-0]     at ClientRequest.emit (node:domain:488:12)
[0-0]     at emitErrorEvent (node:_http_client:101:11)
[0-0]     at Socket.socketErrorListener (node:_http_client:504:5)
[0-0]     at Socket.emit (node:events:519:28)
[0-0]     at Socket.emit (node:domain:488:12)
[0-0]     at emitErrorNT (node:internal/streams/destroy:169:8)
[0-0]     at emitErrorCloseNT (node:internal/streams/destroy:128:3) {
[0-0]   errno: -4078,
[0-0]   code: 'ECONNREFUSED',
[0-0]   syscall: 'connect',
[0-0]   address: '127.0.0.1',
[0-0]   port: 21365
[0-0] }
[0-0]
[0-0] Node.js v20.18.0
[0-0] FAILED in chrome
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### Is there an existing issue for this?

- [X] I have searched the existing issues

**Comment by thamp59** [comment]
Currently working around this on v8 by disabling BiDi and necessary tests that use BiDi

**Comment by christian-bromann** [comment]
@thamp59 thanks for raising this issue. Unfortunately this is a bug within the Sauce Labs platform. Please raise a support ticket with them. Thank you!

**Comment by Emmet-McGuire** [comment]
@thamp59 I was having this same issue a few weeks ago (right around the same time you created this issue) and I had to use `'wdio:enforceWebDriverClassic': true` to get my tests running again on WDIO v9 in saucelabs.  At the time I opened a ticket w/ Saucelabs and they said that was the right fix.

However, I just tried today for the first time w/ `'wdio:enforceWebDriverClassic': false` and things are passing again.

I don't see anything in the Saucelabs release notes indicating they fixed anything, so I am curious if you are seeing it working as well?
